pkg/load/agents/utils.go

```diff
@@ -62,3 +63,72 @@ func startWatchers(path string, callback func() error, recordError func(string))
 	}
 	return nil
 }
+
+func startReleaseRepoWatcher(path string, callback func() error, recordError func(string)) error {
+	errFunc := func(err error, msg string) {
+		recordError(msg)
+		logrus.WithError(err).Error(msg)
+	}
+
+	watcher, err := GetSymlinkWatcher(callback, errFunc, path)
+	if err != nil {
+		return fmt.Errorf("failed to get the symblink watcher: %w", err)
+	}
+	interrupts.Run(watcher)
+
+	return nil
+}
+
+// GetSymlinkWatcher returns a function that watches if the destination of a single symbolic link changed and run
+// the event function against the new destination.
+func GetSymlinkWatcher(eventFunc func() error, errFunc func(error, string), path string) (func(ctx context.Context), error) {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	oldSymlink, err := os.Readlink(path)
+	if err != nil {
+		return nil, err
+	}
+
+	resolveSymlink := func() string { return filepath.Join(filepath.Dir(path), oldSymlink) }
```

Comment:
Can you elaborate more on this one? Do you mean that the `path` variable won't be available inside the closure?

Reply:
`path` does not change, so neither does `filepath.Dir(path)`.  It can be calculated once outside the closure, instead of in each call:

```go
path := "test/test"
f := func() string { return filepath.Dir(path) }
for { f() }
```
```asm
0000000000462220 <main.main>:
…
  462234:       48 8d 05 c1 0e 01 00    lea    rax,[rip+0x10ec1]        # 4730fc <go.string.*+0xbac>
  46223b:       bb 09 00 00 00          mov    ebx,0x9
  462240:       e8 7b fe ff ff          call   4620c0 <path/filepath.Dir>
  462245:       eb ed                   jmp    462234 <main.main+0x14>
…
```